src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -547,6 +560,112 @@ impl EKS {
+                    Option::Some(CommandError::new_from_safe_message(
+                        "Could not find node_group in terra context".to_string(),
+                    )),
+                ))
+            }
+        };
+
+        let region = match RusotoRegion::from_str(&self.region()) {
+            Ok(value) => value,
+            Err(error) => {
+                return Err(EngineError::new_unsupported_region(
+                    event_details,
+                    "".to_string(),
+                    CommandError::new_from_safe_message(error.to_string()),
+                ))
+            }
+        };
+        let credentials = StaticProvider::new(
+            self.cloud_provider.access_key_id(),
+            self.cloud_provider.secret_access_key(),
+            None,
+            None,
+        );
+        let client = Client::new_with(credentials, HttpClient::new().expect("unable to create new Http client"));
+        let eks_client = EksClient::new_with_client(client, region);
+
+        let node_groups = match block_on(eks_client.list_nodegroups(ListNodegroupsRequest {
+            cluster_name: self.cluster_name(),
+            ..Default::default()
+        })) {
+            Ok(res) => res.nodegroups.unwrap_or_default(),
+            Err(_) => return Ok((false, future_node_group.min_nodes)),
+        };
+
+        if node_groups.is_empty() {
+            return Err(EngineError::new_cluster_has_no_worker_nodes(
+                event_details,
+                Option::Some(CommandError::new_from_safe_message(format!(
+                    "Could not find node_groups for cluster {}",
+                    self.cluster_name()
+                ))),
+            ));
+        }
+
+        let actual_nodes_group = match block_on(eks_client.describe_nodegroup(DescribeNodegroupRequest {
+            cluster_name: self.cluster_name(),
+            nodegroup_name: node_groups.first().unwrap().to_string(),
+        })) {
+            Ok(res) => res.nodegroup.unwrap_or_default(),
+            Err(error) => {
+                return Err(EngineError::new_cluster_has_no_worker_nodes(
+                    event_details,
+                    Option::Some(CommandError::new_from_safe_message(error.to_string())),
+                ))
+            }
+        };
+
+        let scaling_config = match actual_nodes_group.scaling_config {
+            Some(value) => value,
+            None => return Ok((false, 0)),
+        };
+
+        let should_update_desired_nodes = scaling_config.min_size.unwrap_or_default()
+            != i64::from(future_node_group.min_nodes)
+            || scaling_config.max_size.unwrap_or_default() != i64::from(future_node_group.max_nodes);
+
+        let kubeconfig = match self.get_kubeconfig_file() {
+            Ok((path, _)) => path,
+            Err(_) => return Ok((false, scaling_config.desired_size.unwrap() as i32)),
+        };
+        let get_node_result = retry::retry(Fixed::from_millis(10000).take(60), || {
```

Comment:
Too huge

Reply:
yep

Reply:
I have changed it to 5, is this better ?